test/boundscheck_exec.jl

```diff
@@ -298,7 +298,7 @@ end
 end |> only === Type{Int}
 
 if bc_opt == bc_default
-@testset "Array/Memory escape analysis" begin
```

Comment:
Why removing the testsets here an in `test/channel_threadpool.jl`?  Other changes look good to me.

Reply:
I explained in the commit message. Basically because they are run in subprocesses their reports are just dumped into stdout in the CI log stream and break up the pattern of the main CI report, adding visual noise. 

I was wondering if we could support passing testesets back from subprocesses somehow but not sure that could work with the current setup.